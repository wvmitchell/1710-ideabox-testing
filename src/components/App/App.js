import React, { Component } from 'react';
import './App.css';
import Form from '../Form/Form'
import CardContainer from '../CardContainer/CardContainer'
class App extends Component {
  constructor(props) {
    super(props) // this gives us this :D
    this.state = {
      ideas: []
    }
  }

  handleSubmit = (idea) => {
    const ideas = [ ...this.state.ideas, idea]
    this.setState({ ideas })
  }

  removeIdea = (id) => {
    const ideas = this.state.ideas.filter(idea => idea.id !== id)
    this.setState({ ideas })
  }

  numIdeas(ideas) {
  }

  render() {
    return (
      <div className="App">
        <h1>My ideas</h1>
        <h2>Num ideas: { this.numIdeas(this.state.ideas) }</h2>
        <Form handleSubmit={this.handleSubmit} />
        <CardContainer ideas={this.state.ideas}
                       removeIdea={this.removeIdea}/>
      </div>
    );
  }
}

export default App;
