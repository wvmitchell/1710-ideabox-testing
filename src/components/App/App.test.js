import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

it('should match the snapshot', () => {
  const renderedComponent = shallow(<App />)
  expect(renderedComponent).toMatchSnapshot()
})

it('when handleSubmit is called with an idea, the state should have that idea', () => {
  const renderedComponent = shallow(<App />)
  const mockedIdea = { title: 'great idea', body: 'learn js', id: 0 }
  renderedComponent.instance().handleSubmit(mockedIdea)
  expect(renderedComponent.state().ideas).toEqual([mockedIdea])
})

it('when removeIdea is called with an id, the state will no longer have the idea with that id', () => {
  const renderedComponent = shallow(<App />)
  const mockedIdea = { title: 'great idea', body: 'learn js', id: 0 }
  renderedComponent.setState({ ideas: [mockedIdea]})
  renderedComponent.instance().removeIdea(0)
  expect(renderedComponent.state().ideas).not.toContain(mockedIdea)
})

it('when i have one idea in state, numIdeas returns 1', () => {
  const renderedComponent = shallow(<App />)
  const mockedIdea = { title: 'great idea', body: 'learn js', id: 0 }
  renderedComponent.setState({ ideas: [mockedIdea]})
  expect(renderedComponent.instance().numIdeas(renderedComponent.state().ideas)).toEqual(1)
})

