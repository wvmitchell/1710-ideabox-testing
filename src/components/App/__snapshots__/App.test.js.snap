// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          My ideas
</h1>,
        <h2>
          Num ideas: 
</h2>,
        <Form
          handleSubmit={[Function]}
/>,
        <CardContainer
          ideas={Array []}
          removeIdea={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My ideas",
        },
        "ref": null,
        "rendered": "My ideas",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Num ideas: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Num ideas: ",
          undefined,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "ideas": Array [],
          "removeIdea": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            My ideas
</h1>,
          <h2>
            Num ideas: 
</h2>,
          <Form
            handleSubmit={[Function]}
/>,
          <CardContainer
            ideas={Array []}
            removeIdea={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My ideas",
          },
          "ref": null,
          "rendered": "My ideas",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Num ideas: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Num ideas: ",
            undefined,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ideas": Array [],
            "removeIdea": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
